src/memory.jl

```diff
@@ -176,6 +176,26 @@ function free(buf::HostBuffer)
     end
 end
 
+##
+# Pinned memory management
+#
+# - IdDict does not free the memory
+# - WeakRef dict does not unique the key by objectid
+# TODO:
+# - Deal with memory regions/views
+const __pinned_memory = Dict{Ptr, WeakRef}()
```

Comment:
@kshyatt I think with using a `Ptr` we no longer need the `WeakRef`.

Reply:
I'll try later today without it